From 41d3b457972bde85991fa7ed6f282370aca4b2af Mon Sep 17 00:00:00 2001
Message-ID: <41d3b457972bde85991fa7ed6f282370aca4b2af.1760476767.git.crobinso@redhat.com>
In-Reply-To: <b825bb556bd3967bf5422c243b77bd4038e317e2.1760476767.git.crobinso@redhat.com>
References: <b825bb556bd3967bf5422c243b77bd4038e317e2.1760476767.git.crobinso@redhat.com>
From: Michal Privoznik <mprivozn@redhat.com>
Date: Fri, 10 Oct 2025 15:20:05 +0200
Subject: [PATCH 2/8] wireshark: Switch header files to #pragma once
Content-type: text/plain

The genxdrstub.pl script generates some header files. But they
use the old pattern to guard against multiple inclusion:

  #ifndef SOMETHING_H
  #define SOMETHING_H
  ...
  #endif

Change the script to generate just '#pragma once' used everywhere
else in our code.

Signed-off-by: Michal Privoznik <mprivozn@redhat.com>
Reviewed-by: Peter Krempa <pkrempa@redhat.com>
Signed-off-by: Cole Robinson <crobinso@redhat.com>
---
 tools/wireshark/util/genxdrstub.pl | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/tools/wireshark/util/genxdrstub.pl b/tools/wireshark/util/genxdrstub.pl
index 8cfda25a27..01b663a88c 100755
--- a/tools/wireshark/util/genxdrstub.pl
+++ b/tools/wireshark/util/genxdrstub.pl
@@ -563,11 +563,8 @@ sub add_header_file {
     local $self->{header_contents} = [];
     $self->print("/* *DO NOT MODIFY* this file directly.\n");
     $self->print(" * This file was generated by $0 from libvirt version $libvirt_version */\n");
-    my $ucname = uc $name;
-    $self->print("#ifndef _$ucname\_H_\n");
-    $self->print("#define _$ucname\_H_\n");
+    $self->print("#pragma once\n");
     $block->();
-    $self->print("#endif /* _$ucname\_H_ */");
     push @{ $self->{headers} }, [ $name, delete $self->{header_contents} ];
 }
 
-- 
2.51.0

